main.tf

```diff
@@ -3,7 +3,11 @@ locals {
   lambda_iam_role_arn   = var.lambda_iam_role_arn != "" ? var.lambda_iam_role_arn : aws_iam_role.lambda[0].arn
   lambda_iam_role_name  = regex(".*role/(?P<role_name>.*)$", local.lambda_iam_role_arn)["role_name"]
   s3_bucket             = var.s3_bucket != "" ? var.s3_bucket : lookup(var.s3_regional_buckets, data.aws_region.current.name, local.default_lambda_bucket)
-  s3_key                = var.s3_key != "" ? var.s3_key : join("/", [var.s3_key_prefix, format("%s.zip", var.lambda_version)])
+  s3_key = var.s3_key != "" ? var.s3_key : (
+    strcontains(var.lambda_version, "/") ?
+    join("/", [var.s3_key_prefix, format("%s.zip", var.lambda_version)]) :
+    format("lambda/observer/arm64/%s.zip", var.lambda_version)
+  )
```

Comment:
cc @bendrucker I had difficulty keeping this logic forward / backwards compatible. Would appreciate if you had a cleaner approach